src/ol/render/canvas/Builder.js

```diff
@@ -306,6 +328,19 @@ class CanvasBuilder extends VectorContext {
         renderer,
         inflateCoordinates,
       ]);
+    } else if (type == GeometryType.MULTI_POINT) {
+      flatCoordinates = geometry.getFlatCoordinates();
+      builderEnd = this.appendFlatPointCoordinates(flatCoordinates, stride);
+      if (builderEnd > builderBegin) {
```

Comment:
This skips the instruction if all points are filtered, though the begin and end geometry instructions are still added.